### DISCUSSION TITLE
Removal of legacy Registry component & CPI v1 API

### DISCUSSION THREAD

**Discussion by jpalermo** [cloudfoundry/bosh#2355] [General]
The registry has been optional and deprecated for a couple of years at this point. We are looking to finish the removal the registry component from the bosh director.

Here is the timeline of what we'd like to do next.

- [x] Remove registry from [bosh-deployment](https://github.com/cloudfoundry/bosh-deployment) for all CPIs that support v2. 
- [x] [bosh-deployment](https://github.com/cloudfoundry/bosh-deployment) no longer deploys registry job
- [x] Remove registry from bosh directors codebase [merged PR](https://github.com/cloudfoundry/bosh/pull/2345)
- [x] Remove registry from bosh cli [merged PR](https://github.com/cloudfoundry/bosh-cli/pull/590)
- [x] Remove CPI v1 from CF maintained CPI codebases (February 2022)
- [x] Remove registry support from bosh-agent (March 2022)
- [x] Remove CPI v1 from Director's codebase (April 2022)
- [x] Clean up bosh.io documentation. All CPI actions documented under CPI v2, CPI v1 clearly marked as removed

We have verified all current CPIs in bosh-deployment support CPI v2 with the exception of SoftLayer, who communicated they were not interested in supporting CPI v2 currently and are aware of the consequences.

**Comment by rkoster** [comment]
@jpalermo do we want to coordinate and merge all these changes at the same time? Or can this PR: https://github.com/cloudfoundry/bosh/pull/2345 for example be merged now since it has been reviewed and approved? Keeping it open till April 2022 could result in a big merge conflict down the road.

**Reply by jpalermo** [reply]
I don't see any reason to hold anything back. It's more of a goal timeline, not a "we won't remove anything until".

Our current belief is that the Registry is no longer needed, so removing any pieces of it should be safe.

**Comment by nouseforaname** [comment]
https://github.com/nouseforaname/bosh-cli/tree/feature/176871612-remove-registry

I've been looking at dropping the registry code from the cli as well.

**Comment by aramprice** [comment]
A related documentation update can be found here https://github.com/cloudfoundry/docs-bosh/pull/763

**Comment by nouseforaname** [comment]
https://github.com/cloudfoundry/bosh-cli/pull/590

The registry code from the CLI has been removed and released in v7 of the bosh cli. This breaks usage with a v1 cpi.

**Comment by nouseforaname** [comment]
https://github.com/cloudfoundry/bosh-agent/pull/287

registry removed from the agent

**Comment by nouseforaname** [comment]
registry dropped from bosh codebase

https://github.com/cloudfoundry/bosh/pull/2345
